scripts/git-all-archive.py

```diff
@@ -0,0 +1,254 @@
+    parser.add_argument('--client_secret', action='store', dest="CLIENTSECRET", help="github Client Secret Oauth Apps")
+
+    args = parser.parse_args()
+
+    RELEASE = args.RELEASE
+    PREFIX = args.PREFIX
+    HELICSTAG = args.HELICSTAG
+    TOKEN = args.TOKEN
+    CLIENTID = args.CLIENTID
+    CLIENTSECRET = args.CLIENTSECRET
+    CLONE = args.CLONE
+    TOPATH = args.TOPATH
+    FILENAME = ""
+    # Clone repository into ./HELICS by default
+    #
+    if CLONE is not None:
+
+        if TOPATH is None:
+            TOPATH = "./HELICS"
+            if os.path.exists(TOPATH):
+                shutil.rmtree(TOPATH)
+        Repo.clone_from(CLONE, TOPATH)
+        repo = Repo(TOPATH)
+    else:
+        if(os.path.exists("./.git")):
+            repo = Repo(".")
+            TOPATH = "."
+        elif (os.path.exists("../.git")):
+            repo = Repo("..")
+            TOPATH = ".."
+        else:
+            print("> Current directory does not contain \".git\"")
+            print("> Are you in \"git\" repository?")
+            print("")
+            parser.print_help(sys.stderr)
+            sys.exit(1)
+
+    if PREFIX is None:
+        PREFIX = "HELICS"
+
+    # Checkout tag or use master
+    #
+    try:
+        if HELICSTAG is None:
+            HELICSTAG = repo.active_branch
+        else:
+            for ref in repo.references:
+                if ref.name == HELICSTAG:
+                    repo.head.reference = ref
+                    HELICSTAG = ref
+    except:
+        print("> Cannot find " + HELICSTAG)
+        print("")
+        parser.print_help(sys.stderr)
+        sys.exit(1)
+
+    # You can push the tarbal in any release tag
+    # By deafult use the HELICSTAG
+    if RELEASE is None:
+        RELEASE = HELICSTAG.name
+
+    if not (isinstance(HELICSTAG, git.refs.tag.TagReference) or isinstance(HELICSTAG, git.refs.head.Head)):
+        print("Could not find tag " + HELICSTAG + " in repository")
+        sys.exit(1)
+
+    # Archive repository and submodules
+    #
+    os.chdir(TOPATH)  # change directory to clone repository
+    currentdir = os.getcwd()
+    print("> archiving main repository")
+    cmd = "git archive --verbose --prefix \"repo/\" --format \"tar\" --output \"" + currentdir + "/" + PREFIX +\
+        "-output.tar\" \"master\""
+    stdoutdata, stderrdata = runcmd(cmd)
+    print("> checkout tag " + HELICSTAG.name)
+    cmd = "git checkout " + HELICSTAG.name
+    stdoutdata, stderrdata = runcmd(cmd)
+    print("> checking out all submodules")
+    cmd = "git submodule update --init"
+    stdoutdata, stderrdata = runcmd(cmd)
+    print("> archiving all submodules")
+    cmd = 'git submodule foreach --recursive \'git archive --verbose --prefix=repo/$path/ --format tar master --output ' +\
+        currentdir + '/repo-output-sub-$sha1.tar\''
+    stdoutdata, stderrdata = runcmd(cmd)
+
+    # Delete pre-existing work
+    #
+    if os.path.exists("/tmp/repo"):
+        shutil.rmtree("/tmp/repo")
```

Comment:
I think it would be better to create a temporary directory using the python function instead of using /tmp, for systems that use different locations for the temp directory (e.g. on one of the clusters TMPDIR is set to /var/tmp/<username> which avoids multiple users trying to use files with the same name): https://docs.python.org/3.7/library/tempfile.html#tempfile.TemporaryDirectory